datasets/api.py

```diff
@@ -111,7 +111,36 @@ def get_queryset(self):
     def perform_create(self, serializer):
         dataset_uuid = self.kwargs['dataset_uuid']
         dataset = Dataset.objects.get(uuid=dataset_uuid)
-        serializer.save(dataset=dataset)
+        user = self.request.user
+        serializer.save(dataset=dataset, last_modified_by=user)
+        dataset.last_modified_by = user
+        dataset.save()
+        if (dataset.scope_content_type.app_label == 'nodes' and
+                dataset.scope_content_type.model == 'instanceconfig'):
+            ic = dataset.scope
+            ic.invalidate_cache()
+
+    def perform_update(self, serializer):
+        user = self.request.user
+        instance = serializer.save(last_modified_by=user)
+        dataset = instance.dataset
+        dataset.last_modified_by = self.request.user
+        dataset.save()
+        if (dataset.scope_content_type.app_label == 'nodes' and
+                dataset.scope_content_type.model == 'instanceconfig'):
+            ic = dataset.scope
+            ic.invalidate_cache()
+
+    def perform_destroy(self, instance):
+        dataset = instance.dataset
+        dataset.last_modified_by = self.request.user
+        dataset.save()
+        if (dataset.scope_content_type.app_label == 'nodes' and
+                dataset.scope_content_type.model == 'instanceconfig'):
+            ic = dataset.scope
+            ic.invalidate_cache()
```

Comment:
A bit of DRY here, maybe just a simple private utility method with `dataset` as the argument?